aopy/visualization/base.py

```diff
@@ -2430,4 +2430,72 @@ def overlay_sulci_on_spatial_map(subject, chamber, drive_type, theta=0, center=(
+    """
+    Plots a 3D plane on a given Matplotlib axis.
+
+    This function generates a square plane centered at the origin, applies a transformation 
+    using a given mapping (either a rotation matrix or a plane equation), and renders it in 3D space.
+
+    Args:
+        ax (mpl_toolkits.mplot3d.Axes3D): The Matplotlib 3D axis on which to plot the plane.
+        plane (numpy.ndarray): Specifies how the plane is transformed:
+            - If shape (3,3) or (4,4): Treated as a transformation matrix for rotation.
+            - If shape (4,): Treated as plane equation coefficients (A, B, C, D) for Ax + By + Cz + D = 0.
+        scale (float, optional): Scaling factor for the plane's size. Default is 1.0.
+        color (str, optional): Color of the plane. Default is 'grey'.
+        alpha (float, optional): Transparency of the plane, where 1 is opaque and 0 is fully transparent. Default is 0.15.
+        resolution (int, optional): Number of subdivisions for the plane. Higher values increase smoothness. Default is 100.
+
+    Raises:
+        ValueError: If 'plane' does not have a valid shape (expected (3,3), (4,4), or (4,)).
+
+    Note:
+        - When 'plane' is a transformation matrix, only the upper-left (3,3) submatrix is used.
+        - When 'plane' is a plane equation (A, B, C, D), the function solves for z using z = (-A * x - B * y - D) / C.
+
+    Examples:
+
+        .. code-block:: python
+
+            import matplotlib.pyplot as plt
+            from mpl_toolkits.mplot3d import Axes3D
+            import numpy as np
+
+            fig = plt.figure()
+            ax = fig.add_subplot(111, projection='3d')
+
+            # Example using a transformation matrix (identity)
+            plane = np.eye(3)  
+            plot_plane(ax, plane, scale=1.0, color='blue', alpha=0.3)
+
+            # Example using a plane equation Ax + By + Cz + D = 0
+            plane_eq = np.array([1, 2, -1, 5])  # x + 2y - z + 5 = 0
+            plot_plane(ax, plane_eq, scale=1.0, color='red', alpha=0.5)
+
+            plt.show()
+
+        .. image:: _images/plot_plane_example.png
+    """
+    
+    xy_range = np.linspace(-10*scale, 10*scale, 100)
+    x, y = np.meshgrid(xy_range, xy_range)
+    
+    # If plane is described as a transformation matrix:
+    if plane.shape in [(3,3),(4,4)]:
+        coords = np.column_stack((x.ravel(), y.ravel(), np.zeros_like(x.ravel())))
+        rotated_coords = coords @ plane[:3, :3]
+        x, y, z = rotated_coords.T.reshape(3, *x.shape)
+    
+    # If plane is described as an equation:
+    elif plane.shape == (4,):
+        A,B,C,D = plane
+        z = (-A * x - B * y - D) / C
+        
+    else:
+        raise ValueError(f"Invalid mapping shape {arr.shape}. Expected (3,3) or (4,4) \
+                         for transformation matrices, or (4,) for plane equations.")
+        
+    ax.plot_surface(x, y, z, alpha=alpha, color=color)
```

Comment:
not necessary but maybe you could add a **kwargs parameter so that if someone wants to adjust some other parameter of plot_surface they can